src/app/core/entity-components/entity-list/list-paginator/list-paginator.component.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, ViewChild, Input, SimpleChanges } from "@angular/core";
+import { Entity } from "../../../entity/entity";
+import { MatPaginator, PageEvent } from "@angular/material/paginator";
+import { MatTableDataSource } from "@angular/material/table";
+import { User } from "app/core/user/user";
+import { SessionService } from "app/core/session/session-service/session.service";
+import { EntityMapperService } from "app/core/entity/entity-mapper.service";
+import { getUrlWithoutParams } from "app/utils/utils";
+import { Router } from "@angular/router";
+
+@Component({
+  selector: "app-list-paginator",
+  templateUrl: "./list-paginator.component.html",
+  styleUrls: ["./list-paginator.component.scss"],
+})
+export class ListPaginatorComponent<E extends Entity> {
+  @Input() dataSource: MatTableDataSource<E>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+
+  user: User;
+  paginatorPageSize: number = 10;
+  paginatorPageSizeBeforeToggle: number = 10;
+  paginatorPageIndex: number = 0;
+  showAllToggle: boolean = false;
+
+  // This key is used to save the pagination settings on the user entity
+  readonly paginatorKey: string;
+
+  constructor(
+    private sessionService: SessionService,
+    private entityMapperService: EntityMapperService,
+    private router: Router,
+  ) {
+    this.paginatorKey = getUrlWithoutParams(this.router);
```

Comment:
There might be a problem when we also use this in the entity subrecord. In case there are multiple tables on one page (e.g. child details) all of them would have the same `paginatorKey` and therefore overwrite each others settings. Can you think of a way to distinguish different tables?